tasks/broad/ImputationTasks.wdl

```diff
@@ -0,0 +1,752 @@
+
+  command {
+    grep -P "SN:~{chrom}\t" ~{ref_dict} | sed 's/.*LN://' | sed 's/\t.*//'
+  }
+  runtime {
+    docker: gatk_docker
+    disks: "local-disk " + disk_size + " HDD"
+    memory: "2 GiB"
+  }
+  output {
+    Int chrom_length = read_int(stdout())
+  }
+}
+
+task GenerateChunk {
+  input {
+    Int start
+    Int end
+    String chrom
+    String basename
+    String vcf
+    String vcf_index
+    Int disk_size = 400 # not sure how big the disk size needs to be since we aren't downloading the entire VCF here
+    String gatk_docker
+  }
+  command {
+    gatk SelectVariants \
+    -V ~{vcf} \
+    --select-type-to-include SNP \
+    --max-nocall-fraction 0.1 \
+    -xl-select-type SYMBOLIC \
+    --select-type-to-exclude MIXED \
+    --restrict-alleles-to BIALLELIC \
+    -L ~{chrom}:~{start}-~{end} \
+    -O ~{basename}.vcf.gz \
+    --exclude-filtered true
+  }
+  runtime {
+    docker: gatk_docker
+    disks: "local-disk " + disk_size + " HDD"
+    memory: "8 GiB"
+  }
+  parameter_meta {
+    vcf: {
+      description: "vcf",
+      localization_optional: true
+    }
+    vcf_index: {
+      description: "vcf index",
+      localization_optional: true
+    }
+  }
+  output {
+    File output_vcf = "~{basename}.vcf.gz"
+    File output_vcf_index = "~{basename}.vcf.gz.tbi"
+  }
+}
+
+task CountChunks {
+  input {
+    File vcf
+    File vcf_index
+    File panel_vcf
+    File panel_vcf_index
+    Int disk_size = ceil(2*size([vcf, vcf_index, panel_vcf, panel_vcf_index], "GiB"))
+    String gatk_docker
+  }
+  command <<<
+    echo $(gatk CountVariants -V ~{vcf}  | sed 's/Tool returned://') > var_in_original
+    echo $(gatk  CountVariants -V ~{vcf} -L ~{panel_vcf}  | sed 's/Tool returned://') > var_in_reference
+  >>>
+  output {
+    Int var_in_original = read_int("var_in_original")
+    Int var_in_reference = read_int("var_in_reference")
+  }
+  runtime {
+    docker: gatk_docker
+    disks: "local-disk " + disk_size + " HDD"
+    memory: "4 GiB"
+  }
+
+}
+task CheckChunks {
+  input {
+    File vcf
+    File vcf_index
+    File panel_vcf
+    File panel_vcf_index
+    Int var_in_original
+    Int var_in_reference
+    Int disk_size =ceil(2*size([vcf, vcf_index, panel_vcf, panel_vcf_index], "GiB"))
+    String bcftools_docker
+  }
+  command <<<
+    if [ $(( ~{var_in_reference} * 2 - ~{var_in_original})) -gt 0 ] && [ ~{var_in_reference} -gt 3 ]; then
+      echo true > valid_file.txt
+      bcftools convert -Ob ~{vcf} > valid_variants.bcf
+      bcftools index -f valid_variants.bcf
+    else
+      echo false > valid_file.txt
+    fi
+  >>>
+  output {
+    File? valid_chunk_bcf = "valid_variants.bcf"
+    File? valid_chunk_bcf_index = "valid_variants.bcf.csi"
+    Boolean valid = read_boolean("valid_file.txt")
+  }
+  runtime {
+    docker: bcftools_docker
+    disks: "local-disk " + disk_size + " HDD"
+    memory: "4 GiB"
+  }
+
+}
+
+task PrePhaseVariantsEagle {
+  input {
+    File? dataset_bcf
```

Comment:
Are these actually optional inputs?

Reply:
no... that's odd.  fixed now.

Reply:
oh, right, it's because the files that get passed to it in the pipeline are optional outputs from a previous task, and in the cases where this is called those outputs always exist.  But I will make the outputs of the previous task non-optional, so that the this task is more self-contained.